finesse/hardware/stepper_motor/st10_controller.py

```diff
@@ -296,14 +296,37 @@ def _relative_move(self, steps: int) -> None:
         self._write_check(f"FL{steps}")
 
     @property
-    def step(self) -> int:
+    def status_code(self) -> int:
+        """The status code of the device.
+
+        For a complete list of status codes and their meanings, consult the manual.
+        """
+        # SC is formatted as a hexadecimal string
+        return int(self._request_value("SC"), 16)
+
+    @property
+    def is_moving(self) -> bool:
+        """Whether the motor is moving.
+
+        This is done by checking whether the status code has the moving bit set.
+        """
+        return self.status_code & 0x0010 == 0x0010
```

Comment:
I don't think I get this. Isn't the part after & always true? And if the order of operations is the opposite, isn't the equality always false, since True or False are never equal to 0x0010? I think I might be missing something.

Reply:
It's a bitwise and operation, not a logical one (i.e. it compares each of the bits of the inputs and returns a new integer). I'm just checking whether this particular bit is set or not.

More info here: https://www.geeksforgeeks.org/check-whether-k-th-bit-set-not/